craft_application/services/lifecycle.py

```diff
@@ -331,6 +332,23 @@ def _get_local_keys_path(self) -> Path | None:
         """
         return None
 
+    def save_pack_state(self, packages: list[str]) -> None:
+        """Persist the names of packed artifacts.
+
+        XXX: This is a PoC shortcut and must be properly reimplemented.
+        """
+        with pathlib.Path("pack_state.txt").open("w") as file:
+            for p in packages:
+                print(p, file=file)
```

Comment:
This is a shortcut that will be replaced when we agree on how to handle application state management.